lib/facility.rb

```diff
@@ -1,14 +1,62 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :hours, :address_id
+  attr_accessor :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
-    @services = []
+  def initialize(details)
+    @name = details[:name]
+    @address = details[:address]
+    @phone = details[:phone]
+    @services = details[:services] || []   
+    @hours = details[:hours] || "Hours not available"
+    @address_id = details[:address_id]
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    return nil unless @services.include?('Vehicle Registration')
+
+    vehicle.registration_date = Date.today
+
+    if vehicle.year <= Date.today.year - 25
```

Comment:
Don't forget about the vehicle methods! You could have called `vehicle.antique?` here ;)